lib/src/conflicts.rs

```diff
@@ -474,7 +512,33 @@ pub fn parse_conflict(input: &[u8], num_sides: usize) -> Option<Vec<Merge<BStrin
     }
 }
 
+/// This method handles parsing both JJ-style and Git-style conflict markers,
+/// meaning that switching conflict marker styles won't prevent existing files
+/// with other conflict marker styles from being parsed successfully. The
+/// conflict marker style to use for parsing is determined based on the first
+/// line of the hunk.
 fn parse_conflict_hunk(input: &[u8]) -> Merge<BString> {
+    // If the hunk starts with a conflict marker, find its first character
+    let initial_conflict_marker_char = input
+        .lines_with_terminator()
+        .next()
+        .filter(|line| is_conflict_marker_line(line))
+        .map(|line| line[0]);
+
+    match initial_conflict_marker_char {
+        // JJ-style conflicts must start with one of these 3 conflict marker lines
+        Some(CONFLICT_DIFF_LINE_CHAR | CONFLICT_MINUS_LINE_CHAR | CONFLICT_PLUS_LINE_CHAR) => {
+            parse_jj_style_conflict_hunk(input)
+        }
+        // Git-style conflicts either must not start with a conflict marker line, or must start with
+        // the "|||||||" conflict marker line (if the first side was empty)
+        None | Some(CONFLICT_GIT_ANCESTOR_LINE_CHAR) => parse_git_style_conflict_hunk(input),
+        // No other conflict markers are allowed at the start of a hunk
+        Some(_) => Merge::resolved(BString::new(vec![])),
+    }
```

Comment:
nit: we might want to handle cases that git-style content had a line identical to jj-style marker and vice versa, but that's not new problem.

(No action requested)